TMDS_3/include/CGAL/IO/File_medit.h

```diff
@@ -865,53 +903,231 @@ output_to_medit(std::ostream& os,
  * \see \ref IOStreamMedit
  */
 template <class C3T3>
 void
 output_to_medit(std::ostream& os,
                 const C3T3& c3t3,
-                bool rebind = false,
-                bool show_patches = false)
+                bool rebind,      // = false,
+                bool show_patches // = false
+#ifndef DOXYGEN_RUNNING
+              , bool all_vertices // = true
+              , bool all_cells    // = false
+#endif
+)
 {
   if ( rebind )
   {
     if ( show_patches )
-      Mesh_3::output_to_medit<C3T3,true,false>(os,c3t3);
+      CGAL::TMDS_3::output_to_medit<C3T3,true,false>(os, c3t3,
+        all_vertices, all_cells);
     else
-      Mesh_3::output_to_medit<C3T3,true,true>(os,c3t3);
+      CGAL::TMDS_3::output_to_medit<C3T3,true,true>(os, c3t3,
+        all_vertices, all_cells);
   }
   else
   {
     if ( show_patches )
-      Mesh_3::output_to_medit<C3T3,false,false>(os,c3t3);
+      CGAL::TMDS_3::output_to_medit<C3T3,false,false>(os, c3t3,
+        all_vertices, all_cells);
     else
-      Mesh_3::output_to_medit<C3T3,false,true>(os,c3t3);
+      CGAL::TMDS_3::output_to_medit<C3T3,false,true>(os, c3t3,
+        all_vertices, all_cells);
   }
 }
 
-
-template<typename T3>
-void write_MEDIT(std::ostream& os, const T3& t3)
+/**
+ * @ingroup PkgTMDS3IOFunctions
+ * @brief outputs a mesh complex to the medit (`.mesh`) file format.
+ *      See \cgalCite{frey:inria-00069921} for a comprehensive description of this file format.
+ * @tparam T3 can be instantiated with any 3D triangulation of \cgal provided that its
+ *  vertex and cell base class are models of the concepts `MeshVertexBase_3` and `MeshCellBase_3`, respectively.
+ * @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+ *
+ * @param os the output stream
+ * @param t3 the triangulation
+ * @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ * \cgalParamNBegin{all_cells}
+ *   \cgalParamDescription{If `true`, all the cells in `t3` are written in `os`,
+ *        however they belong to the complex or not.
+ *        Otherwise, only the cells `c` for which
+ *        `c->subdomain_index() != Subdomain_index()` are written}
+ *   \cgalParamType{Boolean}
+ *   \cgalParamDefault{`false`}
+ *   \cgalParamExtra{This parameter must be set to `true` for the file to be readable by `read_MEDIT()`.}
+ * \cgalParamNEnd
+ *
+ * \cgalParamNBegin{all_vertices}
+ *   \cgalParamDescription{If `true`, all the vertices in `t3` are written in `os`.
+ *                         Otherwise, only the vertices that belong to a cell `c` for which
+ *                         `c->subdomain_index() != Subdomain_index()` are written}
+ *   \cgalParamType{Boolean}
+ *   \cgalParamDefault{`true`}
+ *   \cgalParamExtra{If `all_cells` is `true`, this parameter is ignored. It must be
+ *                   set to `false` for the file to be readable by `read_MEDIT()`.}
```

Comment:
`false`? `all_cells` must be `true` to be readable, and it sets up `all_vertices` to `true` automatically, but  `all_vertices` must be `false` to be readable?